BraveShareTo/ShareToBraveViewController.swift

```diff
@@ -8,15 +8,47 @@ import MobileCoreServices
 import BraveShared
 
 class ShareToBraveViewController: SLComposeServiceViewController {
+    private struct Scheme {
+            private enum SchemeType {
+                case url, query
+            }
+            
+            private let type: SchemeType
+            private let urlOrQuery: String
+            
+            init?(item: NSSecureCoding) {
+                if let text = item as? String {
+                    urlOrQuery = text
+                    type = .query
+                } else if let url = (item as? URL)?.absoluteString.firstURL?.absoluteString {
+                    urlOrQuery = url
+                    type = .url
+                } else {
+                    return nil
+                }
+            }
+            
+            var schemeUrl: URL? {
+                var components = URLComponents()
+                let queryItem: URLQueryItem
+                
+                components.scheme = "brave"
+            
+                switch type {
+                case .url:
+                    components.host = "open-url"
+                    queryItem = URLQueryItem(name: "url", value: urlOrQuery)
+                case .query:
+                    components.host = "search"
+                    queryItem = URLQueryItem(name: "q", value: urlOrQuery)
+                }
+                
+                components.queryItems = [queryItem]
+                return components.url
+            }
```

Comment:
Tried to added URLComponents logic you've mentioned, what do you think?
I also deleted ".addingPercentEncoding" part since encoding is solved by URLComponents type.

Reply:
that's even better, nice find